src/noneP.js

```diff
@@ -0,0 +1,29 @@
+import { compose, map } from 'ramda';
+
+import allP from './allP';
+import rejectP from './rejectP';
+import resolveP from './resolveP';
+
+/**
+ * Returns a Promise that is resolved with an array of reasons when all of the provided Promises reject, or rejected when any Promise is resolved.
+ *
+ * @func noneP
+ * @memberOf RA
+ * @since {@link https://char0n.github.io/ramda-adjunct/2.22.0|v2.22.0}
+ * @category Function
+ * @sig [Promise a] -> Promise [a]
+ * @param {Iterable.<*>} iterable An iterable object such as an Array or String
+ * @return {Promise} A Promise that is resolved with a list of rejection reasons if all Promises are rejected, or a Promise that is rejected with the fulfillment value of the first Promise that resolves.
+ * @example
+ *
+ * RA.noneP([Promise.reject('hello'), Promise.reject('world')]); //=> Promise(['hello', 'world'])
+ * RA.noneP([]); //=> Promise([])
+ * RA.noneP([Promise.reject(), Promise.resolve('hello world')]); //=> Promise('hello world')
+ * RA.noneP([Promise.reject(), 'hello world']); //=> Promise('hello world')
+ */
+const noneP = compose(
```

Comment:
@char0n - I had gone with only 1 map for a slight efficiency improvement, but I agree yours is more readable so I have updated it to match yours. However, in the updated version, I continued using `.then` instead of `thenP` because unfortunately from what I can tell, `thenP` only accepts an `onFulfilled` callback (and not a `onRejected` callback)

Reply:
Yep, make sense. Thanks